src/librustc_parse/parser/mod.rs

```diff
@@ -88,21 +88,10 @@ macro_rules! maybe_recover_from_interpolated_ty_qpath {
 #[derive(Clone)]
 pub struct Parser<'a> {
     pub sess: &'a ParseSess,
-    /// The current non-normalized token.
+    /// The current token.
     pub token: Token,
-    /// The current normalized token.
-    /// "Normalized" means that some interpolated tokens
-    /// (`$i: ident` and `$l: lifetime` meta-variables) are replaced
-    /// with non-interpolated identifier and lifetime tokens they refer to.
-    /// Use this if you need to check for `token::Ident` or `token::Lifetime` specifically,
-    /// this also includes edition checks for edition-specific keyword identifiers.
-    pub normalized_token: Token,
-    /// The previous non-normalized token.
+    /// The previous token.
     pub prev_token: Token,
-    /// The previous normalized token.
-    /// Use this if you need to check for `token::Ident` or `token::Lifetime` specifically,
-    /// this also includes edition checks for edition-specific keyword identifiers.
-    pub normalized_prev_token: Token,
```

Comment:
Consider keeping some of this documentation with a reference to `uninterpolate`. It seems like a good opportunity to remind people who e.g., hover on these fields in VSCode that idents/lifetimes need special care.

Reply:
I'd say that the point of these changes is that idents/lifetimes no longer need special care.
`NtIdent`/`NtLifetime` now need explicit treatment like all other nonterminals, `Token::(ident,uninterpolate,uninterpolated_span)` just provide conveniences for that.

The remaining "special" thing is that `ast::Ident`s must use uninterpolated spans for correct resolution / edition checks, but this looks like an arbitrary place to document that.

Reply:
My main concern is that someone does `if let token::Ident(name, _) = self.token.kind { ... }` which you fixed in this PR and this not being caught in code review. Perhaps this should be documented on the `::Ident(..) | ::Lifetime(..)` variants instead?

Reply:
Yeah, `TokenKind::(Ident,Lifetime)` looks like a better place.